sklearn/manifold/_t_sne.py

```diff
@@ -666,6 +681,15 @@ def _fit(self, X, skip_num_points=0):
             raise ValueError("'method' must be 'barnes_hut' or 'exact'")
         if self.angle < 0.0 or self.angle > 1.0:
             raise ValueError("'angle' must be between 0.0 - 1.0")
+        if self.square_distance not in [True, 'legacy']:
```

Comment:
Don't we also want to allow for `False`?

Otherwise, when we change the default from legacy to True, True will be the only allowed value for `square_distance`.

CC @jeremiedbb

Reply:
Thank you much for your review, here and elsewhere. :)

Just for reference, the relevant comment by @jeremiedbb for the current deprecation plan is [this one](https://github.com/scikit-learn/scikit-learn/issues/12401#issuecomment-652893433).

Reply:
Thank for the ref. Indeed we don't need to allow for False if don't plan on supporting this case in the future.

I don't think that having `0.26: square_distance={True, 'warn'}. square_distance deprecated` will work however: at this point, we will want to deprecate `square_distance=True` and let users leave it to its default (e.g. None) so that their code doesn't break when we completely remove the parameter. But that's an issue for 0.26 and not for this PR.

Reply:
Thank you for the input. I've added a `# TODO` referencing this, at the very least.